src/main/java/org/snomed/snowstorm/ecl/validation/ECLEdgeCaseHandlerService.java

```diff
@@ -0,0 +1,131 @@
+			sEclAttribute.setNumericValue(String.valueOf(Float.MAX_VALUE));
+		} else if (sEclAttribute.getStringValue() != null) {
+			sEclAttribute.setStringValue(MISSING);
+		}
+	}
+
+	private List<SEclAttribute> extractSEclAttributes(final SRefinedExpressionConstraint expressionConstraint) {
+		final SEclRefinement sEclRefinement = (SEclRefinement) expressionConstraint.getEclRefinement();
+		final SSubRefinement subRefinement = (SSubRefinement) sEclRefinement.getSubRefinement();
+		SEclAttributeSet sEclAttributeSet;
+		final List<SEclAttribute> sEclAttributes = new ArrayList<>();
+		if ((sEclAttributeSet = (SEclAttributeSet) subRefinement.getEclAttributeSet()) == null) {
+			sEclAttributeSet = (SEclAttributeSet) subRefinement.getEclAttributeGroup().getAttributeSet();
+			final List<SubAttributeSet> conjunctionAttributeSets = sEclAttributeSet.getConjunctionAttributeSet();
+			final List<SubAttributeSet> disjunctionAttributeSets = sEclAttributeSet.getDisjunctionAttributeSet();
+			if (conjunctionAttributeSets != null || disjunctionAttributeSets != null) {
+				final List<SubAttributeSet> subAttributeSets = new ArrayList<>(conjunctionAttributeSets == null
+						? Collections.emptyList() : conjunctionAttributeSets);
+				subAttributeSets.addAll(disjunctionAttributeSets == null ? Collections.emptyList() : disjunctionAttributeSets);
+				getSEclAttributes(subAttributeSets).stream().filter(Objects::nonNull).forEach(sEclAttributes::add);
+			} else {
+				sEclAttributes.addAll(getSEclAttributes(Collections.singletonList(sEclAttributeSet.getSubAttributeSet())));
+			}
+		} else {
+			sEclAttributes.addAll(getSEclAttributes(Collections.singletonList(sEclAttributeSet.getSubAttributeSet())));
+		}
+		return sEclAttributes;
+	}
+
+	private List<SEclAttribute> getSEclAttributes(final List<SubAttributeSet> subAttributeSets) {
+		final List<SEclAttribute> sEclAttributes = new ArrayList<>();
+		for (final SubAttributeSet subAttributeSet : subAttributeSets) {
+			final SSubAttributeSet sSubAttributeSet = (SSubAttributeSet) subAttributeSet;
+			final SEclAttribute sEclAttribute = (SEclAttribute) sSubAttributeSet.getAttribute();
+			if (sEclAttribute != null) {
+				final SSubExpressionConstraint sSubExpressionConstraint = (SSubExpressionConstraint) sEclAttribute.getAttributeName();
+				if (sSubExpressionConstraint != null && sSubExpressionConstraint.isWildcard()) {
+					return Collections.emptyList();
+				}
+			}
+			sEclAttributes.add(sEclAttribute);
+		}
+		return sEclAttributes;
+	}
+
+	private void cacheConcreteConceptIds(final String branch, final PageRequest pageRequest) {
+		CACHED_CONCRETE_CONCEPT_IDS.addAll(eclQueryService.selectRelevantConceptIds(RETURN_ALL_CONCRETE_ATTRIBUTES_ECL_QUERY,
+				versionControlHelper.getBranchCriteria(branch), branch, false, null, pageRequest, null)
+				.toList().stream().map(Functions.toStringFunction()).collect(Collectors.toList()));
+	}
+
+	@Override
+	public boolean equals(Object o) {
```

Comment:
`@Service` classes should not need to implement `equals` and `hashCode`. These are only needed when the object is put into a collection or compared with another instance of the same class. There should only ever be one instance of a `@Service` class.